### PR TITLE
envoy: 1.25.1 -> 1.25.3

### PR DESCRIPTION
**Pull request by veehaitch**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/http/envoy/default.nix [hunk 1]
```diff
@@ -18,30 +19,23 @@
...
   pname = "envoy";
-  inherit (srcVer) version;
+  version = "1.25.3";
   bazel = bazel_5;
   src = fetchFromGitHub {
     owner = "envoyproxy";
     repo = "envoy";
-    inherit (srcVer) rev;
+    rev = "v${version}";
     sha256 = "sha256-qA3+bta2vXGtAYX3mg+CmSIEitk4576JQB/QLPsj9Vc=";
```
Comment (lukegb) [inline pkgs/servers/http/envoy/default.nix:30]: It seems unlikely that the sha256 of the src repo would not have changed between v1.25.1 and v1.25.3.

**Comment by veehaitch** [comment]
I successfully completed `nix-build -A envoy -A envoy.tests` on `x86_64-linux`. For the CI, however, I always run into a hash mismatch for the FOD. I'm puzzled; maybe you have an idea @lukegb?

**Comment by lukegb** [comment]
Ah - if the FOD is differing on the deps hash that usually means that they've changed something in their build system that generates non-determinism. There's a huge chunk of logic in the envoy derivation to try to find and remove all of it but it doesn't always work.

Unfortunately our current setup for buildBazelPackage is a bit fragile, but it's unclear how we could do much better as it stands.

**Comment by veehaitch** [comment]
Thanks for clarifying! Unfortunately, I'm not very familiar with Bazel let alone `buildBaselPackage`. Therefore, I'm going to close this PR and create a dedicated PR for e6625d599386409a7a604670608a454ba950d028.

**Comment by risicle** [comment]
We're going to have to bump this past 1.25.3 or mark it with `knownVulnerabilities` because of:

https://nvd.nist.gov/vuln/detail/CVE-2023-27488
https://nvd.nist.gov/vuln/detail/CVE-2023-27492
https://nvd.nist.gov/vuln/detail/CVE-2023-27491
https://nvd.nist.gov/vuln/detail/CVE-2023-27496
https://nvd.nist.gov/vuln/detail/CVE-2023-27493

I've tried bumping to 1.25.5 but I get an inconsistent deps hash.

**Pull request closed without merging**